relay-server/src/actors/spooler/mod.rs

```diff
@@ -398,12 +399,18 @@ impl OnDisk {
 
         envelope.set_start_time(start_time.into_inner());
 
-        let managed_envelope = self.buffer_guard.enter(
-            envelope,
-            services.outcome_aggregator.clone(),
-            services.test_store.clone(),
-        )?;
-        Ok(managed_envelope)
+        ProcessingGroup::split_envelope(*envelope)
+            .into_iter()
+            .map(|(group, envelope)| {
```

Comment:
What's the behavior when an old relay attempts to spool an envelope with a group that was added in a more recent version, and we don't fall back to `Unknown`? Maybe I'm getting too paranoid here.

Reply:
@iker-barriocanal how would that happen? Only the serializable envelope is written to disk (not the managed envelope or the group), so the Relay version that reads the envelope from disk decides how it wants to group the envelope. Which is as it should be IMO.

Reply:
Oh right, I assumed it was the managed envelope I guess.